server: TestCheckRestartSafe_Criticality failed

server.TestCheckRestartSafe_Criticality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21303452?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21303452?buildTab=artifacts#/) on release-25.4 @ [3d0ca16a0a0216bc8b53a5eac37928c996140c33](https://github.com/cockroachdb/cockroach/commits/3d0ca16a0a0216bc8b53a5eac37928c996140c33):


```
=== RUN   TestCheckRestartSafe_Criticality
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/04ff595e8b8a4cdf61532aa9937d7edf/logTestCheckRestartSafe_Criticality2195237009
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
*
* WARNING: disk slowness detected: unable to sync log files within 10s
*
    api_v2_test.go:378: 
        	Error Trace:	pkg/server/api_v2_test.go:378
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 78
        	Test:       	TestCheckRestartSafe_Criticality
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/04ff595e8b8a4cdf61532aa9937d7edf/logTestCheckRestartSafe_Criticality2195237009
--- FAIL: TestCheckRestartSafe_Criticality (19.04s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCheckRestartSafe_Criticality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62673

--- a/pkg/server/api_v2_test.go
+++ b/pkg/server/api_v2_test.go
@@ -365,6 +365,7 @@ func TestCheckRestartSafe_Criticality(t *testing.T) {
 
 	testCluster := serverutils.StartCluster(t, 3, base.TestClusterArgs{})
 	defer testCluster.Stopper().Stop(ctx)
+	require.NoError(t, testCluster.WaitForFullReplication())
 
 	ts1 := testCluster.Server(0)
 
@@ -376,7 +377,6 @@ func TestCheckRestartSafe_Criticality(t *testing.T) {
 	require.Greater(t, res.RaftLeadershipOnNodeCount, int32(0))
 	require.Equal(t, int32(1), res.StoreNotDrainingCount)
 
-	require.Equal(t, int32(0), res.UnderreplicatedRangeCount)
 	require.Equal(t, int32(0), res.UnderreplicatedRangeCount)
 
 	err = drain(ctx, ts1, t)

